### ISSUE TITLE
Facebook homepage sign up inputs have extra padding

### ISSUE THREAD

**Issue by jdm** [servo/servo#18831]
Firefox:
<img width="413" alt="screen shot 2017-10-11 at 3 43 34 pm" src="https://user-images.githubusercontent.com/27658/31462836-fe5846f2-ae9b-11e7-8e6a-527a7a78be33.png">

Servo:
<img width="409" alt="screen shot 2017-10-11 at 3 43 21 pm" src="https://user-images.githubusercontent.com/27658/31462484-ee7e969c-ae9a-11e7-9531-93e28e5cc1e7.png">

**Comment by bobthekingofegypt** [comment]
I had a play around with this to try understand what is going on.

There may be more than one issue here but I was looking at the reason for the incorrect height of the input holder, that white space at the bottom.

A simple trigger of the issue is

```
<html>
<head>
<style>
  * {
    background-color: #f7f7f7;
    padding: 0;
    margin: 0;
  }

  .container {
    background-color: red;
  }

  .textbox {
    display:inline-block;
    height: 100px;
    width: 100px;
    background-color: blue;
  }
</style>
</head>
<body>
  <div class="container">
      <div class="textbox"></div>
  </div>
</body>
</html>
```

Display an inline-block inside a block.

![comp1](https://user-images.githubusercontent.com/332141/32348634-5057102e-c00c-11e7-9704-1fca038901b0.png)

What is happening is that during the post style recalc layout pass the code in inline is adding the space required for below baseline text to the calculated height for the div.  This can be emphasised by making the page font size much larger.  

![large-font](https://user-images.githubusercontent.com/332141/32348646-56021082-c00c-11e7-88ff-823e915654f6.png)

Setting font size to 2em, now 30px is added below and extra space is added above, and the blue box is aligned to the baseline of the containers would be text.

I believe this is actually correct behaviour if container has some text in it, but it doesn't.  I think here in lies the problem the layout pass has decided that there was text to be displayed so it added the below baseline space, this effect can be seen in chrome and firefox if text is added to the div after the textbox at which point servo and chrome look identical.

![comp2](https://user-images.githubusercontent.com/332141/32348642-5374e600-c00c-11e7-9e09-fbe44c2a0ed9.png)

I haven't had time to try understand why it thinks there is text or how to turn it off yet, just thought I'd post this incase I don't have time.

**Comment by atouchet** [comment]
Was this actually fixed? I am still able to reproduce the problem.

**Comment by ferjm** [comment]
You are right. I am still able to reproduce the problem with Facebook as well. There's some improvements though.

This is how it looked before #19789:

![before_patch](https://user-images.githubusercontent.com/480202/35263599-b53cbb4a-0019-11e8-83b4-3bf8e6d1dbc0.png)

After #19789:

![after_patch](https://user-images.githubusercontent.com/480202/35263600-b8382488-0019-11e8-9c10-3f602e2a2f94.png)

And how it looks in Firefox:

<img width="420" alt="firefox" src="https://user-images.githubusercontent.com/480202/35263609-c19b7f02-0019-11e8-955a-93a8211f3b2d.png">

The white line at the bottom of the input does not appear anymore after #19789, but the placeholder text inside the input is not centered and sized as it should. I mostly tested the patch with the manual test case so I didn't notice the differences with Facebook's login page.

**Comment by lumiscosity** [comment]
Appears to have been fixed. Tested on Arch Linux x86_64, servo built against commit id 9d29017c0dc6eec4ca7cd868cc13056e0bc0c64a, compared against Firefox 141.0, Epiphany 48.5 and Chromium 138.0.7204.168 (ordered left to right).

<img width="1684" height="398" alt="Image" src="https://github.com/user-attachments/assets/7f54ba02-46e1-4c7f-8f6e-9b04a9e84352" />

<img width="1141" height="740" alt="Image" src="https://github.com/user-attachments/assets/2ed8403f-b1f1-46bb-b739-82f8a11f9a59" />

**Comment by jdm** [comment]
Thanks for verifying!
